### DISCUSSION TITLE
Synchronized functions

### DISCUSSION THREAD

**Discussion by zsmiklos** [simolus3/drift#1117] [Ideas]
Thank you for your efforts and work. Due to asynchronous functions, I can not use Moor. Async is very aggressive and "infects" most of the functions. Async generates many error possibilities in the code when we forget to use the await. Could you possibly create a synchronized version of the functions? It can help many programmers. Thank you in advance.

**Comment by simolus3** [comment]
No, unfortunately this is not possible:

- having both a synchronous and an asynchronous api doubles the maintenance effort
- some implementations, like `moor_flutter` are based on platform channels that _have_ to be async
- advanced features like the isolate api make use of an async communication channel as well

> Async generates many error possibilities in the code when we forget to use the await

The [`unawaited_futures`](https://dart-lang.github.io/linter/lints/unawaited_futures.html) lint sound like it could help.

**Comment by zsmiklos** [comment]
Thank you. It would be good if it were possible

**Comment by kuhnroyal** [comment]
You can't avoid asynchronicity, you have to embrace it.
